### PR TITLE
feat: trigger writing assistant without highlighting text completed

### LINKED ISSUE #349: Trigger writing assistant without highlighting text
### Problem

The writing assistant should be triggerable without having to highlight text in the editor. There are many possible use cases for not wanting to pass in context to the writing assistant.

### Solution
A good implementation I've found is Notion's writing assistant:

https://github.com/user-attachments/assets/3f8240e5-a6fb-45ad-890e-6a09b07bca88

A nice to have feature is a button in the writing assistant for when no text has been highlighted: It'll toggle whether to pass the (full) contents of the current note to the writing assistant or nothing from the current note to the writing assistant.

### PR DESCRIPTION
**Pull request by weilirs**
fix #349 
/claim #349 

https://github.com/user-attachments/assets/82033104-f7ce-4a5e-9a5c-2e159c5dd2b9

### CODE REVIEW

**Review by joseplayero — CHANGES_REQUESTED** [review]
looks nice. made a small comment on the prompt.

some extra thoughts:
- some users may not like seeing the 'press space' text in every new line so maybe we think through ways to allow them to hide it (either in settings or like the message hides after being shown once per note)
- there is a small bug when resizing screen and the message changing places

File: src/components/Writing-Assistant/WritingAssistantFloatingMenu.tsx [hunk 1]
```diff
@@ -189,8 +274,10 @@ const WritingAssistant: React.FC<WritingAssistantProps> = ({
   }
 
   const handleOption = async (option: string, customPromptInput?: string) => {
-    const selectedText = highlightData.text
-    if (!selectedText.trim()) return
+    let selectedText = highlightData.text
+    if (!selectedText.trim() && isSpaceTrigger) {
+      selectedText = ''
```
[the remark this answers is not in the document]
Comment (weilirs) [reply src/components/Writing-Assistant/WritingAssistantFloatingMenu.tsx:279]: I modified how it handles customized prompts, please take a look

**Review by joseplayero — APPROVED** [review]
lg @weilirs 

one thought on this

the backlink extension code could be moved at some point to the EditorManager file because it does very similar things to what you're doing in this PR and moving it would clean up lots of our code (especially in how we pass around the suggestion state )

**Comment by weilirs** [comment]
> lg @weilirs
> 
> one thought on this
> 
> the backlink extension code could be moved at some point to the EditorManager file because it does very similar things to what you're doing in this PR and moving it would clean up lots of our code (especially in how we pass around the suggestion state )

Okay, I'll take a look

**Pull request merged**